lib/mail/elements/address.rb

```diff
@@ -71,6 +71,18 @@ def address(output_type = :decode)
       end
     end
 
+    # Returns the address that is in the address itself with domain IDNA-encoded.
+    #
+    #  a = Address.new('Mikel Lindsaar (My email address) <mikel@tést.lindsaar.net>')
+    #  a.address_idna #=> 'mikel@xn--tst-bma.lindsaar.net'
+    def address_idna
+      if d = domain_idna
+        "#{local}@#{d}"
+      else
+        local
+      end
+    end
```

Comment:
Doesn't this result in two separate cases?  You'll either return the local + domain or just the local.  Wouldn't it be better to just return the local here and put the domain_idna check in the same area where we return the full address (ie, local plus domain) ?

Reply:
I'm not sure I follow. This method just follows the same pattern as `Address#address`, i.e. it does not add `@` if the address does not include a domain.

Reply:
@mikel Would you elaborate? :-)

Reply:
@mikel Ping :-)